packages/plans-grid-next/src/components/shared/header-price/index.tsx

```diff
@@ -123,7 +155,56 @@ const HeaderPrice = ( { planSlug, visibleGridPlans }: HeaderPriceProps ) => {
 		);
 	}
 
-	if ( isAnyVisibleGridPlanOneTimeDiscounted || isAnyVisibleGridPlanOnIntroOffer ) {
+	const termVariantPrice =
+		termVariantPricing?.discountedPrice.monthly ?? termVariantPricing?.originalPrice.monthly ?? 0;
+	const planPrice = discountedPrice.monthly ?? originalPrice.monthly ?? 0;
+	const savings =
+		termVariantPrice > planPrice
+			? Math.floor( ( ( termVariantPrice - planPrice ) / termVariantPrice ) * 100 )
+			: 0;
+
+	if ( enableTermSavingsPriceDisplay && termVariantPricing && savings ) {
+		return (
+			<div className="plans-grid-next-header-price">
+				<div className="plans-grid-next-header-price__badge">
+					{ translate( 'Save %(savings)d%%', {
+						args: { savings },
+						comment: 'Example: Save 35%',
+					} ) }
+				</div>
+				<div
+					className={ clsx( 'plans-grid-next-header-price__pricing-group', {
+						'is-large-currency': isLargeCurrency,
+					} ) }
+				>
+					<PlanPrice
+						currencyCode={ currencyCode }
+						rawPrice={ termVariantPricing.originalPrice.monthly }
+						displayPerMonthNotation={ false }
+						isLargeCurrency={ isLargeCurrency }
+						isSmallestUnit
+						priceDisplayWrapperClassName="plans-grid-next-header-price__display-wrapper"
+						original
+					/>
+					<PlanPrice
+						currencyCode={ currencyCode }
+						rawPrice={ discountedPrice.monthly ?? originalPrice.monthly }
+						displayPerMonthNotation={ false }
+						isLargeCurrency={ isLargeCurrency }
+						isSmallestUnit
+						priceDisplayWrapperClassName="plans-grid-next-header-price__display-wrapper"
+						discounted
+					/>
+				</div>
+			</div>
+		);
+	}
+
+	if (
+		isAnyVisibleGridPlanOneTimeDiscounted ||
+		isAnyVisibleGridPlanOnIntroOffer ||
+		enableTermSavingsPriceDisplay
```

Comment:
Works, but it should be the same as the other two - to only be effective if any plan has a savings.

Reply:
This will tackle in a follow-up. I'm seeing a bit more to clean up around.

Reply:
Doing this and more in https://github.com/Automattic/wp-calypso/pull/96492